src/components/ChatApp/MessageComposer.react.js

```diff
@@ -66,303 +65,278 @@ class MessageComposer extends Component {
     micColor: PropTypes.string,
     focus: PropTypes.bool,
     actions: PropTypes.object,
+    enterAsSend: PropTypes.bool,
+    micInput: PropTypes.bool,
+    messagesByID: PropTypes.object,
+    messages: PropTypes.array,
+    openSnackBar: PropTypes.func,
+    loadingHistory: PropTypes.bool,
   };
 
   constructor(props) {
     super(props);
     this.state = {
       text: props.dream ? `dream ${props.dream}` : '',
-      start: false,
-      stop: false,
-      open: false,
-      result: '',
-      animate: false,
+      isListening: false,
+      isSpeechDialogOpen: false,
+      speechToTextOutput: '',
       rows: 1,
-      recognizing: false,
       micAccess: false,
-      currentArrowIndex: -1, // current message indexed at -1, latest message at 0
+      currentMessageIndex: -1,
+      speechRecognitionTextcolor: '#000',
     };
-    this.flag = 1;
+    this.userMessageHistory = [];
   }
 
-  componentDidMount() {
-    const micInputSetting = UserPreferencesStore.getMicInput();
-    if (micInputSetting) {
+  componentWillMount() {
```

Comment:
cWM lifecycle is deprecated in React latest versions. Please write this block in cDM or constructor as per requirement.

Reply:
Didn't know that !! I have moved the check in the constructor now. @akshatnitd